src/Adapter/Cart/QueryHandler/GetCartForOrderCreationHandler.php

```diff
@@ -127,15 +127,22 @@ public function handle(GetCartInformation $query): CartInformation
             ->setLanguage($language)
             ->setCustomer(new Customer($cart->id_customer));
 
-        $legacySummary = $cart->getSummaryDetails(null, true);
         $addresses = $this->getAddresses($cart);
 
+        if ($query->separateGiftProducts()) {
```

Comment:
If I recall we mentioned two different options the one for gift splitting and the one for free shipping

However I only see one option here, and in the legacy code it's the same you have either:
- `getRawSummaryDetails` : plain cart summary without modifications
- `getSummaryDetails` legacy function which split gifts AND hides the free shipping

I'm fine with only supporting one option for both feature, although it should be validated by @eternoendless, but then we should find a more appropriate name Following the method from legacy we could introduce `$query->getRawSummary()` which default value would then be `true` (and temporarily used with false here)

Reply:
Oh and from what I see in `extractShippingFromLegacySummary` it seems some modifications are also performed in this handler